src/components/RankingAnyTagItem.tsx

```diff
@@ -1,22 +1,39 @@
-import styled from '@emotion/styled';
+import StyledRankingAnyTagItem from '@/styles/StyledRankingAnyTagItem';
+import { useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
 
 const anyTagItemlist = ['받고 싶어한', '많이 선물한', '위시로 받은'];
 
-const StyledRankingAnyTagItem = styled.div`
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  padding: ${({ theme }) => theme.spacing.spacing4};
-  color: ${({ theme }) => theme.palette.blue600};
-`;
-
 const RankingAnyTagItem = () => {
+  const navigate = useNavigate();
+  const { search } = useLocation();
+  const [selected, setSelected] = useState<string | null>('받고 싶어한');
+
+  useEffect(() => {
+    const params = new URLSearchParams(search);
+    const value = params.get('anyTagSelected');
+    if (value) {
+      setSelected(value);
+    }
+  }, [search]);
+
+  const handleClick = (value: string) => {
+    const params = new URLSearchParams(search);
+    params.set('anyTagSelected', value);
+    navigate(`?${params.toString()}`, { replace: true });
+  };
+
   return (
     <>
       {anyTagItemlist.map((item: string) => {
         return (
-          <StyledRankingAnyTagItem key={item} className='ranking-any-tag-item'>
-            <p>{item.toLocaleLowerCase()}</p>
+          <StyledRankingAnyTagItem
+            key={item}
+            className='ranking-any-tag-item'
+            onClick={() => handleClick(item)}
+            isSelected={selected === item}
+          >
+            {item.toLocaleLowerCase()}
```

Comment:
toLocaleLowerCase는 로케일에 따라 다르게 동작해서, 단순 소문자 변환이 목적이시라면 toLowerCase() 정도로도 소구할 수 있을 것 같아용! 예기치 않게 사용자 환경에 따라 결과가 달라질 수 있는 상황을 피할 수. ㅣㅆ어용

```suggestion
            {item.toLowerCase()}
```